src/Runner.Worker/StepsRunner.cs

```diff
@@ -274,6 +260,37 @@ public async Task RunAsync(IExecutionContext jobContext)
             }
         }
 
+        /// <summary>
+        /// Attempts to update the DisplayName of the supplied IActionRunner.
+        /// As the "Try..." name implies, this method should never throw an exception.
+        /// Returns true if the DisplayName is already present or it was successfully updated.
+        /// </summary>
+        private bool TryUpdateDisplayName(IActionRunner actionRunner, out bool updated)
+        {
+            updated = false;
+
+            // REVIEW:  This try/catch can be removed if some future implementation of TryEvaluateDisplayName and UpdateTimelineRecordDisplayName
+            // can make reasonable guarantees that they won't throw an exception.
+            try
+            {
+                if (actionRunner != null &&
+                    actionRunner.Stage == ActionRunStage.Main &&
+                    actionRunner.TryEvaluateDisplayName(actionRunner.ExecutionContext.ExpressionValues, actionRunner.ExecutionContext, out updated))
+                {
+                    if (updated)
+                    {
+                        actionRunner.ExecutionContext.UpdateTimelineRecordDisplayName(actionRunner.DisplayName);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Trace.Warning("Caught exception while attempting to evaulate/update the step's DisplayName.  Exception Details:  {0}", ex);
```

Comment:
now a warning since I can't assume anything about the caller of this new method.